File: ruby/space-age/space_age.rb
class SpaceAge
  def initialize(seconds)
    @seconds = seconds
  end

  def on_earth
    seconds/ORBITAL_PERIODS_SECONDS[:earth]
  end

  def on_mercury
    seconds/ORBITAL_PERIODS_SECONDS[:mercury]
  end

  def on_venus
    seconds/ORBITAL_PERIODS_SECONDS[:venus]
  end

  def on_mars
    seconds/ORBITAL_PERIODS_SECONDS[:mars]
  end

  def on_jupiter
    seconds/ORBITAL_PERIODS_SECONDS[:jupiter]
  end

  def on_saturn
    seconds/ORBITAL_PERIODS_SECONDS[:saturn]
  end

  def on_uranus
    seconds/ORBITAL_PERIODS_SECONDS[:uranus]
  end

  def on_neptune
    seconds/ORBITAL_PERIODS_SECONDS[:neptune]
  end

  private
    attr_reader :seconds
end


ORBITAL_PERIODS_SECONDS = {
                            earth: 31_557_600.to_f,
                            mercury: 31_557_600 * 0.2408467.to_f,
                            venus: 31_557_600 * 0.61519726.to_f,
                            mars: 31_557_600 * 1.8808158.to_f,
                            jupiter: 31_557_600 * 11.862615.to_f,
                            saturn: 31_557_600 * 29.447498.to_f,
                            uranus: 31_557_600 * 84.016846.to_f,
                            neptune: 31_557_600 * 164.79132.to_f
                          }

module BookKeeping
  VERSION = 1
end

File: python/space-age/space_age.py
class SpaceAge(object):
    def __init__(self, seconds):
        self.seconds = seconds

    def on_mercury(self):
        return round(self.seconds/self.orbitalPeriodSeconds("mercury"), 2)

    def on_mercury(self):
        return round(self.seconds/self.orbitalPeriodSeconds("mercury"), 2)

    def on_venus(self):
        return round(self.seconds/self.orbitalPeriodSeconds("venus"), 2)

    def on_earth(self):
        return round(self.seconds/self.orbitalPeriodSeconds("earth"), 2)

    def on_mars(self):
        return round(self.seconds/self.orbitalPeriodSeconds("mars"), 2)

    def on_jupiter(self):
        return round(self.seconds/self.orbitalPeriodSeconds("jupiter"), 2)

    def on_saturn(self):
        return round(self.seconds/self.orbitalPeriodSeconds("saturn"), 2)

    def on_uranus(self):
        return round(self.seconds/self.orbitalPeriodSeconds("uranus"), 2)

    def on_neptune(self):
        return round(self.seconds/self.orbitalPeriodSeconds("neptune"), 2)

    def orbitalPeriodSeconds(self, key):
        orbitalPeriods = {
                            "earth": 31557600,
                            "mercury": 31557600 * 0.2408467,
                            "venus": 31557600 * 0.61519726,
                            "mars": 31557600 * 1.8808158,
                            "jupiter": 31557600 * 11.862615,
                            "saturn": 31557600 * 29.447498,
                            "uranus": 31557600 * 84.016846,
                            "neptune": 31557600 * 164.79132
                          }

        return orbitalPeriods[key]
